src/core/storage/sframe_interface/unity_sframe.cpp

```diff
@@ -101,8 +101,8 @@ void unity_sframe::construct_from_sframe_index(std::string location) {
 
   auto status = fileio::get_file_status(location);
 
-  if ((status.first == fileio::file_status::FS_UNAVAILABLE ||
-       status.first == fileio::file_status::MISSING) &&
+  // web protocol is a bad name, it should mean read-only protocol
```

Comment:
This doesn't seem like the right fix to me; status shouldn't give `REGULAR_FILE` for a directory path over http(s). Is that the behavior you're seeing for URLs to SFrames?

Reply:
But it actually works. Under the hood, it will try to figure out the instance behind the protocol is a directory contains sframe.idx or it's a pure file. If you read the `get_file_status`, it will blindly tell you the URI points to a file.

Reply:
```
tc.SFrame('https://static.turi.com/datasets/audioscrobbler')
Downloading https://static.turi.com/datasets/audioscrobbler/dir_archive.ini to /var/folders/n9/wscz46nj4_10nz3f2y2r0pnr0000gn/T/turicreate-guihaoliang/6732/02d09699-2bdf-4164-8ac2-bb578c36cf1a.ini
Downloading https://static.turi.com/datasets/audioscrobbler/objects.bin to /var/folders/n9/wscz46nj4_10nz3f2y2r0pnr0000gn/T/turicreate-guihaoliang/6732/fd45f74e-7604-41a2-9fcd-3993d0a9c734.bin
Downloading https://static.turi.com/datasets/audioscrobbler/m_74b0dc51.frame_idx to /var/folders/n9/wscz46nj4_10nz3f2y2r0pnr0000gn/T/turicreate-guihaoliang/6732/07f16828-af8d-45ea-b2c0-37d672b77823.frame_idx
Downloading https://static.turi.com/datasets/audioscrobbler/m_74b0dc51.sidx to /var/folders/n9/wscz46nj4_10nz3f2y2r0pnr0000gn/T/turicreate-guihaoliang/6732/e9e00719-1c4c-4cc1-92f5-5efc39fcb005.sidx
Out[3]: Downloading https://static.turi.com/datasets/audioscrobbler/m_74b0dc51.0000 to /var/folders/n9/wscz46nj4_10nz3f2y2r0pnr0000gn/T/turicreate-guihaoliang/6732/e23d74e9-c13b-413e-a131-88daec109dce.0000
Downloading https://static.turi.com/datasets/audioscrobbler/m_74b0dc51.0001 to /var/folders/n9/wscz46nj4_10nz3f2y2r0pnr0000gn/T/turicreate-guihaoliang/6732/db7664e2-2c6e-41b3-955b-cd6975223b1b.0001
Downloading https://static.turi.com/datasets/audioscrobbler/m_74b0dc51.0002 to /var/folders/n9/wscz46nj4_10nz3f2y2r0pnr0000gn/T/turicreate-guihaoliang/6732/47fc0d15-ae41-4c1b-a742-6097155fb7f7.0002
Downloading https://static.turi.com/datasets/audioscrobbler/m_74b0dc51.0003 to /var/folders/n9/wscz46nj4_10nz3f2y2r0pnr0000gn/T/turicreate-guihaoliang/6732/d199c616-56a6-463f-a55a-c131a51bc3c1.0003
Downloading https://static.turi.com/datasets/audioscrobbler/m_74b0dc51.0004 to /var/folders/n9/wscz46nj4_10nz3f2y2r0pnr0000gn/T/turicreate-guihaoliang/6732/caf93341-386d-4563-a63b-8c03c6ef02db.0004
Downloading https://static.turi.com/datasets/audioscrobbler/m_74b0dc51.0005 to /var/folders/n9/wscz46nj4_10nz3f2y2r0pnr0000gn/T/turicreate-guihaoliang/6732/4a6e7edd-abeb-4e7e-9dd7-e63c655c96a5.0005
Downloading https://static.turi.com/datasets/audioscrobbler/m_74b0dc51.0006 to /var/folders/n9/wscz46nj4_10nz3f2y2r0pnr0000gn/T/turicreate-guihaoliang/6732/36221993-abd7-4b8c-ae2f-08f944a03f2b.0006
Downloading https://static.turi.com/datasets/audioscrobbler/m_74b0dc51.0007 to /var/folders/n9/wscz46nj4_10nz3f2y2r0pnr0000gn/T/turicreate-guihaoliang/6732/30c0d6aa-0b05-4b60-8647-f0a9e9f68e0a.0007

Columns:
        user_id str
        item_id str
        target  int

Rows: 418036

Data:
+---------+----------------+--------+
| user_id |    item_id     | target |
+---------+----------------+--------+
| 1001606 | Raver's Nature |   2    |
| 1002598 | Raver's Nature |   1    |
| 1009768 | Raver's Nature |   29   |
| 1020855 | Raver's Nature |   9    |
| 1024677 | Raver's Nature |   1    |
| 1025593 | Raver's Nature |   3    |
| 1036036 | Raver's Nature |   51   |
| 1037250 | Raver's Nature |   3    |
| 1037372 | Raver's Nature |   1    |
| 1037666 | Raver's Nature |   1    |
+---------+----------------+--------+
[418036 rows x 3 columns]
Note: Only the head of the SFrame is printed.
You can use print_rows(num_rows=m, num_columns=n) to print more rows and columns.
```

Reply:
I get that it works, but to me that is very strange and something doesn't seem right. The directory path doesn't actually exist at that URL -- it should give a 404 not found, which I would think would translate into a `MISSING` rather than `REGULAR_FILE`. Is it possible the bug is deeper in the fileio code?

Reply:
oh, right. I forget to push the other code. Just a minute.

Reply:
nope, the GitHub folds that section:
```
if (fileio::try_to_open_file(location + "/dir_archive.ini")) {
      status.first = fileio::file_status::DIRECTORY;
      status.second.clear();
    }
```
If it's non writeable remote file, it doesn't do further ado to make sure it exist in remote or not. So here, we have extra logic to try to figure out it means a directory or a file.

Reply:
You might need extra credentials to see the dir. For me, I cannot curl the URL neither.

Reply:
@znation -- this was triggered by the https being a thin wrapper to an s3 service, so it ends up not being a directory because s3 doesn't have directories, just buckets and keys.

Reply:
@hoytak Thanks for the explanation!